src/rgw/rgw_perf_counters.cc

```diff
@@ -128,11 +128,17 @@ CountersContainer get(req_state *s) {
 
   if (user_counters_cache && !s->user->get_id().id.empty()) {
     std::string key = ceph::perf_counters::key_create(rgw_op_counters_key, {{"User", s->user->get_id().id}});
+    if (!s->user->get_tenant().empty()) {
+      key = ceph::perf_counters::key_insert(key, {{"Tenant", s->user->get_tenant()}});
```

Comment:
is there a reason we shouldn't add Tenant unconditionally? this seems like a case where empty label values are useful. the Tenant label means you could query prometheus for all counters under a given `Tenant=foo` - would you be able to query `Tenant=` the same way if we don't add the empty label?

p.s. the `key_insert()` would cost another string allocation here, when passing a possibly-empty Tenant to `key_create()` wouldn't

Reply:
You should be able to query `Tenant=foo` as long as `foo` exists as a tenant even if we didn't add the empty `Tenant` label.

My reasoning for making tenant optional was:
1. I'm not sure how widely used tenant is. If a user doesn't know about the concept of a tenant, I wouldn't want to cause confusion when reading the labels.
2. If tenant is not enabled then every op metric would have have an empty tenant label. Though the number of entries in Prometheus would be the same, I'm didn't want to add however small cost of the additional label would be in Prometheus.
 
However I don't feel too strongly about my reasoning and would add tenant unconditionally.

Reply:
Also does https://github.com/ceph/ceph/pull/53003/commits/d6d111d750625713acd55971fa0063257575635a do the trick for reducing string allocations if the tenant is left optional?

Reply:
that helps, yeah. please just make sure this handling of tenants is documented